petroflow/src/well.py

```diff
@@ -367,3 +368,137 @@ def drop_short_segments(self, min_length):
+            If `True`, values will be aggregate.
+            If `False`, values will be concatenate.
+
+        Returns
+        -------
+        numpy.ndarray
+            Assembled array.
+        """
+        func = 'mean' if func != 'max' else 'max'
+        if getattr(self.segments[0], '_'+attr) is None:
+            return None
+        pixels_per_m = self.segments[0].pixels_per_cm * 100
+
+        agg_array_hight_pix = round((self.depth_to-self.depth_from)*pixels_per_m)
+        attr_val_shape = getattr(self.segments[0], '_'+attr).shape
+
+        total = np.zeros((agg_array_hight_pix, *attr_val_shape[1:]), dtype=int)
+        background = np.full_like(total, np.nan, dtype=np.double)
+        for segment in self.segments:
+            attr_val = getattr(segment, '_'+attr)
+            segment_place = slice(round((segment.depth_from-self.depth_from)*pixels_per_m),
+                                  round((segment.depth_to-self.depth_from)*pixels_per_m))
+
+            if aggregate is False:
+                background[segment_place] = attr_val
+                continue
+            if func == 'max':
+                background[segment_place] = np.fmax(background[segment_place], attr_val)
+                continue
+            if func == 'mean':
+                background[segment_place] = np.nansum([background[segment_place], attr_val], axis=0)
+                total[segment_place] += 1
+
+        if func == 'max' or aggregate is False:
+            return background
+        if func == 'mean':
+            total = np.where(total == 0, 1, total)
+            return background / total
+        raise ValueError("Aggregation function can't be ({})".format(func))
+
+    def aggregate(self, func, attrs_to_aggregate=None, level=None):
+        """Aggregate or concatenate loaded segments attributes from
+        `WellSegment.attrs_fdtd_index`, `WellSegment.attrs_depth_index`,
+        `WellSegment.attrs_pixel_index` into one segment. This segment's
+        `depth_from` and `depth_to` will be minimum `depth_from`
+        and maximun `depth_to` for all aggregated segments.
+
+        Parameters
+        ----------
+        func : str, callable or list
+            Function to use for aggregating the data.
+            - `str` - short function name (e.g. ``'max'``, ``'min'``, ``'sum'``,
+               ``'mean'``, ``'count'``, ``'var'``, ``'std'``, ``'size'``, ``'sem'``).
+            - `callable` - a function which gets a `pd.Series` and returns
+               `float` or `int` value.
+            - `list` of `str` and/or `callable` described above.
+        attrs_to_aggregate : list of str, optional
+            Names of attributes which will be aggregate by the `func` function.
+            Other attributes will be concatenate.
+            Default is ``['logs', 'core_uv', 'core_dl']``.
+        level : int, optional
+            Level of the well tree defined for aggregation.
+            Default is minus the depth of the tree.
```

Comment:
```suggestion
            Defaults to an aggregation of the whole tree.
```
If you can explain default behaviour instead of default value - do it.